### ISSUE TITLE
Foreign members is broken

### ISSUE THREAD

**Issue by patricoferris** [geocaml/ocaml-topojson#3]
Currently the foreign members implementation is not correct as it filters a global set of `keys_in_use` see:

https://github.com/geocaml/ocaml-topojson/blob/38c765e7ed539f83b554af9feb7858be098eef0e/src/topojson/topojson.ml#L29-L39

These shouldn't be applied to every geometry. For example a `Point` may contain `"coordinates"` but `"arcs"` should be considered a foreign member for a point. The code currently doesn't do this.

**Comment by Priceless-P** [comment]
Hi @patricoferris ,

My name is Prisca Chidimma. I am one of the Outreachy Applicants. I want to contribute to this project. I just got my environment set up. I am not very familiar with OCaml but I am more than willing to learn. I will take some time to go through the material provided on the project page and i will work on this issue.

**Comment by dinakajoy** [comment]
Sure @Priceless-P, you can go ahead and work on this issue. It's great that you have your setup ready, feel free to drop comments when blocked.

**Comment by Priceless-P** [comment]
Okay @dinakajoy, thank you very much.

**Comment by Priceless-P** [comment]
Hello Mentor @patricoferris,  Please how can I compile and run the scripts in my local machine so that I can see if the changes I made are producing the desired results?

**Comment by patricoferris** [comment]
I (very recently) added a contributing guide which should help with this https://github.com/geocaml/ocaml-topojson/blob/main/CONTRIBUTING.md

In terms of testing this issue, I think you could use the example I described above

> For example a Point may contain "coordinates" but "arcs" should be considered a foreign member for a point. The code currently doesn't do this.

So in our tests we could add `"arcs"` to the point to make sure it gets parsed as a foreign member and not as an actual "key in use". For clarity, this point:

https://github.com/geocaml/ocaml-topojson/blob/c9e7bcd57e0126edf340f63c535d758d7a0c707a/test/topojson/test_cases/files/exemplar.json?short_path=32c13b9#L7-L13

**Comment by Priceless-P** [comment]
Okay @patricoferris , But I am still having a little issue understanding the concept of foreign members. Does it mean that if  `arcs`  is a foreign member to  `point` for example, it shouldn't be in the `point` object?

**Comment by patricoferris** [comment]
Yep, they are JSON key-value pairs that are not in the TopoJSON spec. The explanation in thee GeoJSON RFC might help https://www.rfc-editor.org/rfc/rfc7946#section-6.1

**Comment by Priceless-P** [comment]
@patricoferris @dinakajoy. I have been looking at this code for a while and the best idea I have come up with is to find a way of getting the values of `assoc` here 
```let foreign_members_of_json json =
      match J.to_obj json with
      | Ok assoc ->
          List.filter (fun (k, _v) -> not (List.mem k keys_in_use)) assoc
      | Error _ -> []
```
then check if it is `==  Point` or `MutiPoint` then skip if so. Is this approach correct? Can you recommend any resource I can read up that will enable me solve this issue?

**Comment by patricoferris** [comment]
That sounds like it is heading in the right direction. I think you've highlighted the important point -- we need a means to distinguish between when we have a `Point` vs. when we have a `Multipoint` etc. I think we can push the foreign_member calculation into the functions called `X.base_of_json` so they can be specific to `X`. E.g. this call

https://github.com/geocaml/ocaml-topojson/blob/c9e7bcd57e0126edf340f63c535d758d7a0c707a/src/topojson/topojson.ml#L274

could also return the foreign members for a point.

**Comment by Priceless-P** [comment]
Oh yes.  That makes a lot of sense. Thanks! @patricoferris. I will try that now.

**Comment by olybeke** [comment]
hello @patricoferris, 
I'm Oluchi Nwankwo, an outreachy applicant and I would love to know if I can also work on this issue. I have done my Ocaml environment setup and I'm ready to contribute. I just want to ensure I can be assigned since the issue isn't fixed yet.

**Comment by dinakajoy** [comment]
@Priceless-P are you still working on this issue?
Please confirm

**Comment by Priceless-P** [comment]
Yes I am @dinakajoy. I am almost done with it. Have even created a PR. I just need to add the test and that will be all.

**Comment by patricoferris** [comment]
Fixed in #15
